src/MongoDB.Driver/Linq/Translators/FindProjectionTranslator.cs

```diff
@@ -218,12 +218,9 @@ private static IReadOnlyList<IFieldExpression> GetUniqueFieldsByHierarchy(IEnume
                 var referenceGroup = referenceGroups.Dequeue();
                 if (!skippedFields.Contains(referenceGroup.Key))
                 {
-                    var hierarchicalReferenceGroups = referenceGroups
-                        .Where(x =>
-                        {
-                            return x.Key.Count(c => c == '.') != referenceGroup.Key.Count(c => c == '.') && // otherwise, fields are on the same nesting level
-                                   x.Key.StartsWith(referenceGroup.Key);
-                        });
+                    var prefix = referenceGroup.Key + '.';
```

Comment:
NOTE: 
with this line:
```
referenceGroup.Key + '.'
```
checking on the number of `dot`s is not obligatory.

NOTE2: 
Linq queries similar to this one (without a `dot` in one from fields): 
```
Project(p => new {  F = p, a = p.C }, "{ C : { D : '1', E1 : { F : 2 } } }")
```

which have a document itself doesn't generate `Projection` field. See this test: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Tests/Linq/Translators/FindProjectionTranslatorTests.cs#L47
So, they are correct.